aiidalab_widgets_base/viewers.py

```diff
@@ -15,12 +18,23 @@ def viewer(obj, downloadable=True, **kwargs):
     :type downloadable: bool
 
     Returns the object itself if the viewer wasn't found."""
+    import inspect
+    import warnings
+    from aiida.orm import Node
+
+    if not (inspect.isclass(type(obj)) and issubclass(type(obj), Node)):  # only working with AiiDA nodes
+        warnings.warn("This viewer works only with AiiDA objects, got {}".format(type(obj)))
+        return obj
 
     try:
         _viewer = AIIDA_VIEWER_MAPPING[obj.node_type]
         return _viewer(obj, downloadable=downloadable, **kwargs)
-    except (AttributeError, KeyError):
-        return obj
+    except (KeyError) as exc:
+        if obj.node_type in str(exc):
+            warnings.warn("Did not find an appropriate viewer for the {} object. Returning the object "
+                          "itself.".format(type(obj)))
```

Comment:
At least one of the two warnings appears redundant. Maybe combine the two warnings?

Reply:
Why? The first warning is about a non-aiida object, the second one tells that for the given AiiDA object the viewer is not implemented